src/client/sandbox/node/live-node-list/wrapper.js

```diff
@@ -1,64 +1,35 @@
 import nativeMethods from '../../native-methods';
 import { isShadowUIElement } from '../../../utils/dom';
-
-const arrayFilter = Array.prototype.filter;
+import WrapperState from './wrapper-state';
 
 export default class LiveNodeListWrapper {
-    constructor (nodeList, domContentLoadedEventRaised, tagName) {
-        nativeMethods.objectDefineProperty.call(window.Object, this, 'item', {
-            value: index => {
-                this._refreshNodeList();
+    constructor (nodeList, tagName) {
+        const state = new WrapperState(nodeList, tagName);
 
-                return this._filteredNodeList[index];
-            }
-        });
-        nativeMethods.objectDefineProperty.call(window.Object, this, 'length', {
-            get: () => {
-                this._refreshNodeList();
+        nativeMethods.objectDefineProperties.call(window.Object, this, {
+            item: {
+                value: index => {
+                    state.refreshNodeListIfNecessary();
 
-                return this._filteredNodeList.length;
-            }
-        });
+                    return state.filteredNodeList[index];
+                }
+            },
 
-        if (this.namedItem) {
-            nativeMethods.objectDefineProperty.call(window.Object, this, 'namedItem', {
-                value: (...args) => {
-                    const findNamedItem = this._nodeList.namedItem.apply(this._nodeList, args);
+            namedItem: {
```

Comment:
Use the old way for this
![image](https://user-images.githubusercontent.com/4133518/31117311-0ce65eb8-a832-11e7-87be-e5ce09fc12bc.png)